### DISCUSSION TITLE
v3: Refactor live reload config

### DISCUSSION THREAD

**Discussion by tafelnl** [ionic-team/capacitor#3217] [Ideas]
Considering live reload: https://capacitorjs.com/docs/guides/live-reload

I do not think it is very good practice to have the live reload server config within the capacitor.config.json

As the docs states itself:
> Be careful not to commit the server config to source control.

When you are just developing on your own, you will most definitely forget it at least once. Let alone when you are developing in small teams, or even on enterprise scale.

I am not sure how this could be solved as I am quite a developer rookie so to say, but I definitely see problems with this in our development team in our office.

Maybe some sort of `.env`-ish approach ??

**Comment by tafelnl** [comment]
Ah I see now that on the [Capacitor 3.0 issue](https://github.com/ionic-team/capacitor/issues/3133) it says

> Dynamic Capacitor Config. Capacitor should allow a capacitor.config.{ts,js} file so that configuration can be generated based on environment. Capacitor would run this during sync and generate a capacitor.config.json as normal in each native platform. Resolves #1741, #1478.

So it is already being considered. Excuse me for not doing my research well enough

**Reply by imhoffd** [reply]
Thanks for bringing it up! I'm still not sure it makes sense in `capacitor.config.ts` because everyone would have the same conditional, e.g.:

```typescript
{
  ...otherOptions,
  server: process.env.HOST ? { url: `${process.env.HOST}:${process.env.PORT ?? 8100 }` : undefined
}
```

Part of evaluating the dynamic config will be re-addressing live reload configuration, so I'm sure we'll come up with something. Thanks again!
